content/article/passing-by.md

```diff
@@ -0,0 +1,237 @@
+{
+   "image" : "/images/passing-by/relay.jpg",
+   "thumbnail" : "/images/passing-by/relay.jpg",
+   "date" : "2021-07-13T11:48:52",
+   "categories" : "development",
+   "title" : "Passing by",
+   "tags" : [
+      "reference",
+      "functions"
+   ],
+   "draft" : false,
+   "description" : "Looking at how arguments are passed to subs",
+   "authors" : [
+      "thibault-duponchelle"
+   ]
+}
+
+## Assignments by copy or reference
+When assigning, `perl` often copies the value, see the following code:
+```perl
+my $var1 = "not modified";
+
+my $var2 = $var1; # Copy the content, not the variable itself
+$var1 = "modified";
+
+print "$var2\n";
+```
+This code will print the initial *non-modified* value:
+```
+not modified
+```
+
+It is because the assignment copies the content of the variable and does not *alias* the variable. 
+
+To better understand, you can check by printing the [reference](https://perldoc.perl.org/perlref) of the variable:
+```perl
+# Compare storage location
+print \$var1 . " vs " . \$var2 . "\n";
+```
+That gives 2 different storage locations:
+```perl
+SCALAR(0x55589c6378e0) vs SCALAR(0x55589c6378f8)
+                   ^^                        ^^  
+```
+So it clearly states that these are 2 different variables.
+
+But wait, there is also some cases where the assignment is actually *aliasing*, it is what happen in `foreach` loops:
+```perl
+my @array = ( "foo", "bar", "baz" );
+
+foreach my $var (@array) {
+    $var = "modified"; # Modify the initial @array
+}
+
+print "@array\n";
+```
+And the array is well modified:
+```
+modified modified modified
+```
+
+## A word about references
+A [reference](https://perldoc.perl.org/perlref) represents a single scalar value that holds *where-the-value-is-stored* informations.
+Having 2 different locations means 2 different variables.
```

Comment:
This isn't really true though. You don't need variables at all to have two different locations.

Reply:
OK thank you I have to find another way to say this